apis/core/lib/domain/queries/csv-source.ts

```diff
@@ -17,3 +17,29 @@ export async function sourceWithFilenameExists(csvMapping: NamedNode, fileName:
       `
     .execute(client.query)
 }
+
+export async function getSourcesFromMapping(csvMapping: NamedNode, client = streamClient): Promise<any> {
+  const stream = await SELECT.DISTINCT`?source`
+    .WHERE`      
+    GRAPH ${csvMapping}
+    {
+      ${csvMapping} ${cc.csvSource} ?source
+    }
+    `
+    .execute(client.query)
+
+  return new Promise((resolve, reject) => {
+    const sources: string[] = []
+
+    stream.on('data', row => {
+      Object.entries(row).forEach(([, value]) => {
+        if (value) {
+          sources.push(value as string)
+        }
+      })
+    })
+
+    stream.on('end', () => resolve(sources))
+    stream.on('error', error => reject(error))
+  })
+}
```

Comment:
This can be much simpler with a `parsingClient` (also exported from `../../query-client`)

```suggestion
export async function * getSourcesFromMapping(csvMapping: NamedNode, client = parsingClient) {
  const results = await SELECT.DISTINCT`?source`
    .WHERE`
    GRAPH ${csvMapping}
    {
      ${csvMapping} ${cc.csvSource} ?source
    }
    `
    .execute(client.query)

  for (const result of results) {
    const source = result.source
    if (source.termType === 'NamedNode') {
      yield source
    }
  }
}
```

Reply:
Yes this is nice, but I go with the other suggestion :wink: